From 98a22f5fcb68054d26051ea9ca67bca17497c451 Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 6 May 2020 00:01:28 +0530
Subject: [PATCH 1356/1851] mtd: spi-nor: Add 4byte opcode support for Macronix
 flash parts

commit b9a93ada8b13db84a9f712ce3e789dff4c5ec558 from
https://github.com/Xilinx/linux-xlnx.git

Added 4byte opcode support for MX66L2G45G and MX66U2G45G Macronix flash parts.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index bc078d712916..d7dc7fcd281b 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2628,9 +2628,11 @@ static const struct flash_info spi_nor_ids[] = {
 	{ "mx66u1g45g",  INFO(0xc2253b, 0, 64 * 1024, 2048, SECT_4K |
 			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ "mx66l2g45g",  INFO(0xc2201c, 0, 64 * 1024, 4096, SECT_4K |
-			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+			SPI_NOR_4B_OPCODES) },
 	{ "mx66u2g45g",  INFO(0xc2253c, 0, 64 * 1024, 4096, SECT_4K |
-			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+			SPI_NOR_4B_OPCODES) },
 	{ "mx66l1g55g",  INFO(0xc2261b, 0, 64 * 1024, 2048, SPI_NOR_QUAD_READ) },
 
 	/* Micron <--> ST Micro */
-- 
2.31.1

